modules/ROOT/pages/fd-iterating.adoc

```diff
@@ -0,0 +1,112 @@
+
+.This diagram represents a flow in which a Parallel For-Each component is used. The rounded rectangles are cards.
+image::fd-parallel-foreach.png[Diagram depicting the actions taken by the Parallel For-Each component]
+
+<1> A card passes the payload of a flow to the Parallel For-Each component. The payload consists of a collection of elements.
+<2> The Parallel For-Each component divides the collection into separate elements. It then sends elements simultaneously to the first component, connector, or module that is in its scope. You can specify the maximum number of elements to process at a time. Each element is treated as a payload within the scope. So, if a connector within the scope is configured to act on `payload` or uses a script to process `payload`, the payload is not the payload that the Parallel For-Each component received.
+<3> After all of the elements are processed, the Parallel For-Each component aggregates and passes the results of the processing that was done within its scope. The results form the payload that is passed to the next card in the flow.
+<4> The next card in the flow receives the results from the Parallel For-Each component, and the flow continues.
+
+The order of the results correspond to the order of the elements in the original payload. For example, suppose that the payload that is passed to a Parallel For-Each component is `["Apple","Banana","Cherry"]`. Within the component's scope is a Set Payload component that processes each element as a separate payload. The Set Payload component processes each payload with this DataWeave expression: `payload ++ '-result'`. The Parallel For-Each component passes these results to the next card in the flow: `["Apple-result","Banana-result","Cherry-result"]`.
+
+[IMPORTANT]
+====
+Within the scope of the Parallel For-Each component, the order in which each element is processed is not guaranteed to be the order in which each element appears in a collection. Therefore, if any of the connectors require elements in a particular order (for example, if you want to update database tables in a particular order with the Database connector), you should use the For-Each component instead.
+
+Moreover, you should be aware whether any APIs or services that you want to connect to within the scope of a Parallel For-Each component have any limits in the number of concurrent connections that they accept. If they do, you must specify a maximum number of concurrent processes for the Parallel For-Each component.
+====
+
+
+== Transactional Processing
+
+The For-Each component does nothing differently when processing a transaction.
+
+However, the Parallel For-Each component does not process the elements of a collection simultaneously when processing a transaction. Instead, it processes them serially, just as the For-Each component does. The output from this component is still the aggregated results from the processing of the elements in a collection.
```

Comment:
What do we mean when we talk about a transaction? I'm not familiar with the wording.
Can we provide an example here?

Reply:
I've removed this section for now. I'm also not yet familiar with what a "transaction" means in this context.